### ISSUE TITLE
Bolt4 login FAILs if *interactive* `user-add` is used to create user/credentials

### ISSUE THREAD

**Issue by huna02** [bolt/core#2472]
Bolt Version: 4.1.15
Install type: Composer install
PHP version: 7.4.16
Web server: nginx 1.19.8

Bolt4 login FAILs
```
Argument 1 passed to Symfony\Component\HttpFoundation\RedirectResponse::__construct() must be of the type string, null given, called in /tmp/bolt/vendor/bolt/core/src/Security/LoginFormAuthenticator.php on line 147
```

if *interactive* `user-add` is used to create user/credentials

clean-install of
```
cd /tmp
rm -rf bolt
composer -n create-project bolt/project bolt
cd  bolt

cat .env
	APP_ENV=dev
	APP_DEBUG=1
	APP_SECRET=67d829bf61dc5f87a73fd814e2c9f629
	TRUSTED_PROXIES=127.0.0.1,10.1.10.17
	TRUSTED_HOSTS='10.1.10.17|bolt\.example\.com|dev\.example\.com|frontend\.example\.com'
	DATABASE_URL=sqlite:////tmp/bolt.db
	MAILER_DSN=smtp://localhost
	CORS_ALLOW_ORIGIN=^https?://(localhost|127\\.0\\.0\\.1)(:[0-9]+)?$

rm -f /tmp/bolt.db
bin/console doctrine:database:create

bin/console doctrine:schema:create
bin/console doctrine:fixtures:load --no-interaction

_USERNAME="my_admin"
_PASSWORD="CHANGEME"
_EMAIL="my_admin@example.com"
_DISPLAYNAME="my_admin"

bin/console bolt:add-user --admin ${_USERNAME} ${_PASSWORD} ${_EMAIL} ${_DISPLAYNAME}
composer run post-create-project-cmd
composer update
bin/console make:migration --no-interaction
bin/console doctrine:migrations:migrate --no-interaction
```

@ nav to

	https://bolt.example.com/bolt

login with credentials^ FAILs
```
Argument 1 passed to Symfony\Component\HttpFoundation\RedirectResponse::__construct() must be of the type string, null given, called in /tmp/bolt/vendor/bolt/core/src/Security/LoginFormAuthenticator.php on line 147
```

repeating above procedure, but switching to NON-INTERACTIVE add-user
```
-	bin/console bolt:add-user --admin ${_USERNAME} ${_PASSWORD} ${_EMAIL} ${_DISPLAYNAME}
+	bin/console  bolt:add-user --admin
```

then, @ login, with same credentials, works

**Comment by I-Valchev** [comment]
Hi @huna02 ,

Thanks for the detailed bug report. Unfortunately, I cannot reproduce the steps. I am sorry, but I don't understand exactly what happened at each step to reproduce.

Can you explain?

**Comment by huna02** [comment]
(1) create a bolt project
```
cd /tmp
rm -rf bolt
composer -n create-project bolt/project bolt
cd  bolt
```

(2) populate .env with DB credentials etc
```
cat << EOF > .env
APP_ENV=dev
APP_DEBUG=1
APP_SECRET=67d829bf61dc5f87a73fd814e2c9f629
TRUSTED_PROXIES=127.0.0.1,10.1.10.17
TRUSTED_HOSTS='10.1.10.17|bolt\.example\.com|dev\.example\.com|frontend\.example\.com'
DATABASE_URL=sqlite:////tmp/bolt.db
MAILER_DSN=smtp://localhost
CORS_ALLOW_ORIGIN=^https?://(localhost|127\\.0\\.0\\.1)(:[0-9]+)?$
EOF
```

(3) create & populated the DB
```
rm -f /tmp/bolt.db
bin/console doctrine:database:create
bin/console doctrine:schema:create
bin/console doctrine:fixtures:load --no-interaction

```

(4) define admin credentials
```
_USERNAME="my_admin"
_PASSWORD="CHANGEME"
_EMAIL="my_admin@example.com"
_DISPLAYNAME="my_admin"
```

(5) NON-INTERACTIVELY create the admin user
```
bin/console bolt:add-user --admin ${_USERNAME} ${_PASSWORD} ${_EMAIL} ${_DISPLAYNAME}
```

(6) update & migrate
```
composer run post-create-project-cmd
composer update
bin/console make:migration --no-interaction
bin/console doctrine:migrations:migrate --no-interaction
```

(7) navigate to Bolt login and enter those^ credentials

Bolt4 login FAILs with ERROR:

	"Argument 1 passed to Symfony\Component\HttpFoundation\RedirectResponse::__construct() must be of the type string, null given, called in /tmp/bolt/vendor/bolt/core/src/Security/LoginFormAuthenticator.php on line 147"


next, REPEAT exactly the same procedure, EXCEPT do NOT

	(5) NON-INTERACTIVELY create the admin user
	```
	bin/console bolt:add-user --admin ${_USERNAME} ${_PASSWORD} ${_EMAIL} ${_DISPLAYNAME}
	```

instead,

	(5) INTERACTIVELY create the admin user
	```
	bin/console bolt:add-user --admin
		...
	```

proceed to login, and Bolt4 login SUCCEEDSs with NO error

**Comment by I-Valchev** [comment]
@huna02 hanks! can you verify, once you are able to login, if the user that is created from the environment variables has the correct username and email?

I don't know if underscores are allowed that way, at least they don't work inside the `.env` file when I tested it.

**Comment by bobdenotter** [comment]
This should be fixed in `master` in #2476 and #2501
